pkg/kinflate/util/multitransformer.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// MultiTransformer contains a list of transformers.
+type MultiTransformer struct {
+	Transformers []Transformer
```

Comment:
Lets continue to avoid public members.

This object needs a constructor that returns an immutable object, meaning the incoming slice should be copied.  something like

```
func NewMultiTransformer(arg []Transformer) Transformer {
	r := &MultiTransformer{
		transformers: make([]Transformer, len(arg))}
	copy(r.transformers, arg)
	return r
}
```
alternatively, one could have an Append method, but no reason to have  both.

Reply:
yes, Append would make the object mutable, but in a simple obvious way - more constrained than having an underlying slice that could change at any time.